File: src/bin/0136-single-number.py
# @leetup=info id=136 lang=python3 slug=single-number



# @leetup=code
class Solution:
    def singleNumber(self, nums: list[int]) -> int:
        a = 0
        for n in nums:
            a ^= n
        return a


# @leetup=code

File: src/bin/0136-single-number.cpp
// @leetup=info id=136 lang=cpp slug=single-number

#include <algorithm>
#include <cassert>
#include <iostream>
#include <map>
#include <set>
#include <tuple>
#include <vector>

// @leetup=code

class Solution {
 public:
  int singleNumber(std::vector<int>& nums) {
    int res = 0;
    for (int n : nums) {
      res ^= n;
    }
    return res;
  }
};
// @leetup=code

int main() {
  std::cout << "Hello, world!" << std::endl;
  return 0;
}
